src/PersonObjects/PersonMethods.ts

```diff
@@ -181,7 +181,7 @@ export function updateSkillLevels(this: Person): void {
     Math.floor(this.calculateSkill(this.exp.charisma, this.mults.charisma * currentNodeMults.CharismaLevelMultiplier)),
   );
   //In case MaxHP is Infinite, setting ratio to 1
-  const ratio: number = Math.min(this.hp.current / this.hp.max, 1) || 1;
+  const ratio = Number.isFinite(this.hp.current) ? this.hp.current / this.hp.max : 1;
```

Comment:
You lost the Math.min here (looking back, it looks like there are/were edge-cases where HP could go over max and this is needed to deal with that).